lib/screens/home/products/pages/product_modal_screen.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:test_proj/models/index.dart';
+import 'package:test_proj/router/index.dart';
+import 'package:test_proj/screens/home/products/pages/product_modal_form_bloc.dart';
+import 'package:test_proj/screens/home/products/stx_products_bloc.dart';
+import 'package:test_proj/services/index.dart';
+import 'package:test_proj/widgets/form_builder/index.dart';
+
+@RoutePage()
+class ProductModalScreen extends StatelessWidget implements AutoRouteWrapper {
+  const ProductModalScreen({
+    this.product,
+    super.key,
+  });
+
+  final Product? product;
+
+  @override
+  Widget wrappedRoute(BuildContext context) {
+    return BlocProvider(
+      create: (context) => getIt<ProductModalFormBloc>(param1: product),
+      child: this,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final formBloc = context.read<ProductModalFormBloc>();
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(
+          formBloc.isEditing ? 'Edit the product' : 'Add a new product',
+        ),
+      ),
+      body: CustomFormBlocListener(
+        formBloc: formBloc,
+        onSuccess: (_, state) {
+          // ignore: cast_nullable_to_non_nullable
+          final product = state.response as Product;
+          final productBloc = context.read<StxProductsBloc>();
+
+          if (state.isEditing) {
+            productBloc.editItem(product);
+          } else {
+            productBloc.addItem(product);
+          }
+        },
```

Comment:
It might be better to do this inside the form bloc, and use onSuccess to show the success page or go somewhere else when onSubmit returns a success.

Reply:
it's done on the page so that the form bloc doesn't depend on productBloc